### ISSUE TITLE
Date class doc refers to "year zero"

### ISSUE THREAD

**Issue by nickshanks** [mdn/sprints#3858]
## URL(s)
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#Two_digit_years_map_to_1900_–_1999

## Request type
- [ ] New documentation
- [X] Correction or update

## Details
Text currently reads "In order to create and get dates between the years 0 and 99 the Date.prototype.setFullYear() and Date.prototype.getFullYear() methods should be used."

There is no "year zero". The 31st of December 1 B.C. was followed the next day by the 1st of January 1 A.D. under current reckoning. Consequently, for example, 5 BC was nine years before 5 AD, not ten.

Furthermore, one cannot know for certain whether a value of zero meant the year after 1 BC, or the year before 1 AD, as the developer's intent cannot be predicted without complex heuristics. Passing a value coercible to zero to the function Date.prototype.setFullYear() should thus result in an error (haven't tried doing that, just saying that it should).

I propose amending the text to something along the lines of "In order to create and get dates between the years 1 CE and 99 CE the Date.prototype.setFullYear() and Date.prototype.getFullYear() methods should be used. Passing a value of 0 is invalid, as there was no year zero."

**Comment by Lafi0** [comment]
const moonLanding = new Date('Nov 4, 20 00:20:18');

console.log(moonLanding.getFullYear());

**Comment by visdajpal** [comment]
I think it should be "1900 and 1999" in full

**Comment by MendyBerger** [comment]
Looks to me like this example demonstrates what it says on the `Date` constructor page:

> Values from 0 to 99 map to the years 1900 to 1999. All other values are the actual year. See the example below.

Where do you folks think that this example belongs, on the `Date` main page (where it currently is), or in the `Date` constructor page since this is where the behavior is documented, or maybe even on both?

BTW The current "see example below" link doesn't work since it looks for the example on the constructor page, but the example actually resides on the main `Date` page.

**Comment by MendyBerger** [comment]
I added the text from the Date constructor page to clarify what the example demonstrates, and fixed the link to the example, for now I left the example on the main Date object page.

To address the year zero problem, according to [Wikipedia](https://en.wikipedia.org/wiki/ISO_8601#Years) the ISO 8601 actually does allow the year zero. I can't find any official documentation stating that the `Date` object follows ISO 8601, but I tested `setFullYear(0)` in all major browsers (Chromium, Firefox, WebKit) and they all happily changed the year to 0, so I think its fair to conclude that even though year zero isn't a valid year, this case is an exception.
